logicrunner/builtin/contract/account/account.go

```diff
@@ -108,20 +109,101 @@ func (a *Account) RollBack(amountStr string) error {
 	return nil
 }
 
-// TransferToAccount transfers funds to account.
-func (a *Account) TransferToAccount(amountStr string, toAccount insolar.Reference) error {
-	to := account.GetObject(toAccount)
-	return a.transfer(amountStr, to)
-}
-
 // TransferToDeposit transfers funds to deposit.
 func (a *Account) TransferToDeposit(amountStr string, toDeposit insolar.Reference) error {
 	to := deposit.GetObject(toDeposit)
 	return a.transfer(amountStr, to)
 }
 
+// TransferToMember transfers funds to member.
+func (a *Account) TransferToMember(amountStr string, toMember insolar.Reference) error {
+	to := member.GetObject(toMember)
+	return to.Accept(amountStr)
+}
+
 // GetBalance gets total balance.
 // ins:immutable
 func (a *Account) GetBalance() (string, error) {
 	return a.Balance, nil
 }
+
+// Transfer transfers money to given member.
+func (a *Account) Transfer(rootDomainRef insolar.Reference, amountStr string, toMember *insolar.Reference) (interface{}, error) {
+
+	amount, ok := new(big.Int).SetString(amountStr, 10)
+	if !ok {
+		return nil, fmt.Errorf("can't parse input amount")
+	}
+	zero, _ := new(big.Int).SetString("0", 10)
+	if amount.Cmp(zero) < 1 {
+		return nil, fmt.Errorf("amount must be larger then zero")
+	}
+
+	ccRef := foundation.GetCostCenter()
+
+	cc := costcenter.GetObject(ccRef)
+	feeStr, err := cc.CalcFee(amountStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculate fee for amount: %s", err.Error())
+	}
+
+	fee, ok := new(big.Int).SetString(feeStr, 10)
+	if !ok {
+		return nil, fmt.Errorf("can't parse input feeStr")
+	}
+
+	toFeeMember, err := cc.GetFeeMember()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get fee member: %s", err.Error())
+	}
+
+	totalSum, err := safemath.Add(fee, amount)
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculate totalSum for amount: %s", err.Error())
+	}
+
+	currentBalanceStr, err := a.GetBalance()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get balance for asset: %s", err.Error())
+	}
+	currentBalance, _ := new(big.Int).SetString(currentBalanceStr, 10)
+	if totalSum.Cmp(currentBalance) > 0 {
+		return nil, fmt.Errorf("balance is too low: %s", currentBalanceStr)
+	}
+
+	newBalance, err := safemath.Sub(currentBalance, totalSum)
+	if err != nil {
+		return nil, fmt.Errorf("not enough balance for transfer: %s", err.Error())
+	}
+	a.Balance = newBalance.String()
+
+	err = a.TransferToMember(amountStr, *toMember)
```

Comment:
is this needs to be a saga call?

Reply:
In the last scheme, the saga is called from account to member.
member.Accept(amountStr) market as SAGA